libraries/chain/include/eosio/chain/global_property_object.hpp

```diff
@@ -153,16 +153,23 @@ namespace eosio { namespace chain {
 
    inline snapshot_global_property_object::extension_t get_gpo_extension(const global_property_object& gpo) {
       return snapshot_global_property_object::extension_v0{
-          gpo.proposed_security_group_block_num, gpo.proposed_security_group_participants, gpo.transaction_hooks};
+          gpo.proposed_security_group_block_num,
+          {gpo.proposed_security_group_participants.begin(), gpo.proposed_security_group_participants.end()},
+          {gpo.transaction_hooks.begin(), gpo.transaction_hooks.end()}};
    }
 
    inline void set_gpo_extension(global_property_object&                        gpo,
                                  snapshot_global_property_object::extension_t&& extension) {
       std::visit(
           [&gpo](auto& ext) {
              gpo.proposed_security_group_block_num    = ext.proposed_security_group_block_num;
-             gpo.proposed_security_group_participants = std::move(ext.proposed_security_group_participants);
-             gpo.transaction_hooks                    = std::move(ext.transaction_hooks);
+             gpo.proposed_security_group_participants = {ext.proposed_security_group_participants.begin(),
```

Comment:
I'm not familiar with boost containers but isn't same thing supposed to do copy constructor?

Reply:
They are 2 different types now, so you cannot use the copy constructor

Reply:
cleaning these up by using a set_*** method that just takes begin and end iterators.